pkg/controllermanager/controller/cloudprofile/reconciler.go

```diff
@@ -91,39 +91,5 @@ func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 		}
 	}
 
-	// TODO voelzmo - this migration step ensures that all MachineImageVersions in the Cloud Profile contain `docker` in their list of supported Container Runtimes
-	// This can be removed in a couple of versions. Note that while this is still in here, it is impossible to add an image without `docker` support!
-	migrationHappened := migrateMachineImageVersionCRISupport(cloudProfile)
```

Comment:
Long story short, I will drop the cleanup commit from this PR and open an issue with my findings so that we can clean it up in a dedicated PR without mixing up too many things in this PR.

Reply:
ref https://github.com/gardener/gardener/issues/6375